src/features/search/helpers/useSearchHistory/useSearchHistory.ts

```diff
@@ -0,0 +1,116 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { getHistoryLessThan30Days } from 'features/search/helpers/useSearchHistory/helpers/getHistoryLessThan30Days'
+import { CreateHistoryItem, HistoryItem } from 'features/search/types'
+import { eventMonitoring } from 'libs/monitoring'
+import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
+
+export const HISTORY_KEY = 'search_history'
+export const MAX_HISTORY_RESULTS = 20
+export const MIN_HISTORY_RESULTS = 3
+
+export function useSearchHistory() {
+  const { showErrorSnackBar } = useSnackBarContext()
+  const [history, setHistory] = useState<HistoryItem[]>([])
+  const [queryHistory, setQueryHistory] = useState<string>('')
+
+  const setHistoryItems = useCallback(async (newItems: HistoryItem[]) => {
+    return AsyncStorage.setItem(HISTORY_KEY, JSON.stringify(newItems))
+  }, [])
+
+  const getHistoryFromStorage = useCallback(async () => {
+    try {
+      const items = await AsyncStorage.getItem(HISTORY_KEY)
+
+      if (!items) return []
+
+      const history: HistoryItem[] = JSON.parse(items)
+
+      const historyLessThan30Days = getHistoryLessThan30Days(history)
+      await setHistoryItems(historyLessThan30Days)
+
+      return historyLessThan30Days
+    } catch (error) {
+      return []
+    }
+  }, [setHistoryItems])
+
+  useEffect(() => {
+    const fetchHistory = async () => {
+      const historyData = await getHistoryFromStorage()
+      setHistory(historyData)
+    }
+
+    fetchHistory()
+  }, [getHistoryFromStorage])
```

Comment:
Je n'avais pas fait gaffe mais ça pourrait faire partie des petites améliorations @cedricls1 

```suggestion
  useEffect(() => {
    getHistoryFromStorage().then(setHistory)
  }, [getHistoryFromStorage])
```